batch/batch/log_store.py

```diff
@@ -11,51 +11,69 @@
 
 
 class LogStore:
-    def __init__(self, batch_logs_bucket_name, instance_id, blocking_pool, *, project=None, credentials=None):
+    def __init__(self, batch_logs_bucket_name, instance_id, *, credentials=None):
         self.batch_logs_bucket_name = batch_logs_bucket_name
         self.instance_id = instance_id
         self.batch_logs_root = f'gs://{batch_logs_bucket_name}/batch/logs/{instance_id}/batch'
-        self.gcs = GCS(blocking_pool, project=project, credentials=credentials)
+
+        self.gcs_fs = GoogleStorageAsyncFS(credentials=credentials)
+        self.sem = asyncio.Semaphore(50)
 
         log.info(f'BATCH_LOGS_ROOT {self.batch_logs_root}')
         format_version = BatchFormatVersion(BATCH_FORMAT_VERSION)
         log.info(f'EXAMPLE BATCH_JOB_LOGS_PATH {self.log_path(format_version, 1, 1, "abc123", "main")}')
 
+    async def _write(self, path, data, binary=True):
+        async with await self.gcs_fs.create(path) as f:
+            if not binary:
+                data = data.encode('utf-8')
+            await f.write(data)
+
+    async def _rmtree(self, url):
+        async with self.sem:
+            await self.gcs_fs.rmtree(self.sem, url)
+
+    async def _open_from(self, url, start, end, binary=True):
+        async with await self.gcs_fs.open_from(url, start, end) as f:
+            data = await f.read()
+            if not binary:
+                data = data.decode('utf-8')
+            return data
+
     def batch_log_dir(self, batch_id):
         return f'{self.batch_logs_root}/{batch_id}'
 
     def log_path(self, format_version, batch_id, job_id, attempt_id, task):
         if not format_version.has_attempt_in_log_path():
             return f'{self.batch_log_dir(batch_id)}/{job_id}/{task}/log'
-
         return f'{self.batch_log_dir(batch_id)}/{job_id}/{attempt_id}/{task}/log'
 
     async def read_log_file(self, format_version, batch_id, job_id, attempt_id, task):
-        path = self.log_path(format_version, batch_id, job_id, attempt_id, task)
-        return await self.gcs.read_gs_file(path)
+        url = self.log_path(format_version, batch_id, job_id, attempt_id, task)
+        return await self.gcs_fs.open(url)
```

Comment:
open returns a file doesn't it? I think we should `.read()` and convert from bytes to a string